pkg/agent/run.go

```diff
@@ -141,6 +141,12 @@ func run(ctx context.Context, cfg cmds.Agent, proxy proxy.Proxy) error {
 		}
 	}
 
+	notifySocket := os.Getenv("NOTIFY_SOCKET")
+	os.Unsetenv("NOTIFY_SOCKET")
+
+	os.Setenv("NOTIFY_SOCKET", notifySocket)
```

Comment:
This should be unset from the environment while starting child processes (kubelet, containerd, etc) so that they don't try to notify for uys, and then restored again before calling `systemd.SdNotify` since it looks to this variable for the notification socket path.